### PR TITLE
Added spring wrapper implementations for Resource, Fixed manifest for osgi

### PR DESCRIPTION
**Pull request by garpinc**
Added spring wrapper implementations for spring (hence upgraded dependent version), Fixed manifest to properly create the correct osgi artifacts

### CODE REVIEW

**Comment by garpinc** [comment]
Seems 1.8.8 pac4j needs to be released before this will build successfully. Builds fine against 1.8.8-SNAPSHOT locally.

File: pom.xml [hunk 1]
```diff
@@ -57,7 +57,7 @@
         <spring.security.version>4.0.3.RELEASE</spring.security.version>
         <spring.version>4.2.2.RELEASE</spring.version>
         <servlet.version>3.0.1</servlet.version>
-        <pac4j.version>1.8.7</pac4j.version>
+        <pac4j.version>1.8.8</pac4j.version>
```
Comment (leleuj) [inline pom.xml]: I doubt you could have tested it locally: it should be `1.8.8-SNAPSHOT`.

File: pom.xml [hunk 2]
```diff
@@ -116,6 +116,34 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
```
Comment (leleuj) [inline pom.xml]: These dependencies appear twice.

File: src/main/java/org/pac4j/springframework/config/ResourceWrapper.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import org.pac4j.core.io.Resource;
+
+/**
+ * A wrapper that adapts a spring {@link org.springframework.core.io.Resource} to a pac4j {@link Resource}.
+ * 
+ * @author Keith Garry Boyce
+ * @since 1.4.3
+ */
+public class ResourceWrapper implements Resource {
```
Comment (leleuj) [inline src/main/java/org/pac4j/springframework/config/ResourceWrapper.java]: OK. Now I have the big picture. I think we could need that in the `spring-webmvc-pac4j` or in `j2e-pac4j` libraries for example.

So you should move these two classes in your pull requests (in the `io`) package, rename them as `SpringResourceWrapper` and `Spring WritableResourceWrapper` with the Spring dependencies added as `optional` in pac4j.

File: pom.xml (hunk 1 above)
Comment (garpinc) [inline pom.xml]: It works locally with that in Travis it didn't which is why it is what  it is now

File: src/main/java/org/pac4j/springframework/config/ResourceWrapper.java (hunk 3 above)
Comment (garpinc) [inline src/main/java/org/pac4j/springframework/config/ResourceWrapper.java]: Got it will fix this tomorrow morning

File: pom.xml (hunk 1 above)
Comment (leleuj) [inline pom.xml]: The current 1.8.x branch of pac4j builds a 1.8.8-SNAPSHOT version, not a 1.8.8.

**Comment by leleuj** [comment]
Resuming the review on this as https://github.com/pac4j/pac4j/pull/539 has been merged.

**Comment by leleuj** [comment]
Can you upgrade this library to use pac4j v1.8.8-SNAPSHOT?

File: pom.xml [hunk 4]
```diff
@@ -167,13 +167,52 @@
                 <version>3.0.1</version>
                 <configuration>
                     <instructions>
-                        <Bundle-SymbolicName>org.pac4j.core</Bundle-SymbolicName>
-                        <Export-Package>org.pac4j.springframework.security.*;version=${project.version}</Export-Package>
+                        <Bundle-SymbolicName>org.pac4j.springframework.security</Bundle-SymbolicName>
+                        <Export-Package>org.pac4j.springframework.config;version=${project.version}, org.pac4j.springframework.security.*;version=${project.version}</Export-Package>
```
Comment (leleuj) [inline pom.xml]: We don't need the `org.pac4j.springframework.config` package any more.

File: pom.xml [hunk 5]
```diff
@@ -167,13 +167,52 @@
...
                     </instructions>
                 </configuration>
             </plugin>
         </plugins>
+        <pluginManagement>
+			<plugins>
+			    <plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-jar-plugin</artifactId>
+					<version>2.5</version>
```
Comment (leleuj) [inline pom.xml]: Can you use the latest version if it's not the case?

File: pom.xml [hunk 6]
```diff
@@ -167,13 +167,52 @@
...
+							<manifestFile>
+								${project.build.outputDirectory}/META-INF/MANIFEST.MF
+							</manifestFile>
+						</archive>
+					</configuration>
+				</plugin>
+				<plugin>
+					<groupId>org.apache.felix</groupId>
+					<artifactId>maven-bundle-plugin</artifactId>
+					<version>2.5.0</version>
```
Comment (leleuj) [inline pom.xml]: Can you use the latest version if it's not the case?

**Comment by leleuj** [comment]
A few comments before the merge. Thanks

**Comment by garpinc** [comment]
I think I addressed your comments..

**Comment by leleuj** [comment]
Perfect

**Pull request merged**